jetstream/analysis.py

```diff
@@ -924,15 +1193,159 @@ def run(
                             period,
                         ).model_dump(warnings=False)
 
-                    segment_results.root += self.counts(
-                        segment_data, segment, analysis_basis
-                    ).model_dump(warnings=False)
+                        segment_results.root += self.counts(
+                            segment_data, segment, analysis_basis
+                        ).model_dump(warnings=False)
 
+                    # save statistics for this analysis basis
+                    result = self.save_statistics(
+                        segment_results.model_dump(warnings=False),
+                        self._table_name(
+                            period.value, len(time_limits.analysis_windows), statistics=True
+                        ),
+                    )
+                    results.append(result)
+                    stats_results.append(result)
+                else:
+                    # convert metric configurations to mozanalysis metrics
+                    summary_metrics: list[Summary] = [
+                        m
+                        for m in self.config.metrics[period]
+                        if (
+                            m.metric.analysis_bases == analysis_basis
+                            or analysis_basis in m.metric.analysis_bases
+                        )
+                        and m.metric.select_expression is not None
+                    ]
+                    config_metrics: set[Metric] = {
+                        Metric.from_metric_config(m.metric).to_mozanalysis_metric()
+                        for m in summary_metrics
+                        if m.metric.name in metric_slugs
+                    }
+
+                    sanity_metrics: set[Metric] = set()
+                    # get sanity metrics for each data source
+                    # for metric in config_metrics:
+                    #     ds: DataSource | None = metric.data_source
+                    #     if ds is not None:
+                    #         sanity_metrics.update(
+                    #             ds.get_sanity_metrics(self.config.experiment.normandy_slug)
+                    #         )
+
+                    # create the full list of metrics to compute
+                    metrics = config_metrics | sanity_metrics
+                    logger.debug(f"COMPUTING METRICS: {[m.name for m in metrics]}")
+                    for metric in metrics:
```

Comment:
this will run in parallel with dask, right?

Reply:
Yep, I checked the task graph with a run and it all looks parallel to me.